File: speedviewlib/src/main/java/com/github/anastr/speedviewlib/util/OnPrintTickLabel.java
package com.github.anastr.speedviewlib.util;

/**
 * <p>
 *     callback to draw custom TickLabel for each Tick.
 * </p>
 * this Library build By <b>Anas Altair</b>
 * see it on <a href="https://github.com/anastr/SpeedView">GitHub</a>
 */

public interface OnPrintTickLabel {
    /**
     * @param tickPosition position of ticks, start from 0.
     * @param tick speed value at the tick.
     * @return label to draw.
     */
    CharSequence getTickLabel(int tickPosition, float tick);
}

File: speedviewlib/src/main/java/com/github/anastr/speedviewlib/util/OnPrintTickLabel.kt
package com.github.anastr.speedviewlib.util

/**
 *
 *
 * callback to draw custom TickLabel for each Tick.
 *
 * this Library build By **Anas Altair**
 * see it on [GitHub](https://github.com/anastr/SpeedView)
 */

interface OnPrintTickLabel {
    /**
     * @param tickPosition position of ticks, start from 0.
     * @param tick speed value at the tick.
     * @return label to draw.
     */
    fun getTickLabel(tickPosition: Int, tick: Float): CharSequence
}
